api_tutorials/bait_cnn/bait_performance_cnn.py

```diff
@@ -0,0 +1,605 @@
+
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# ~~~~~~~ Models ~~~~~~~~~~~~~~~~~~
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+def build_small_sequential_bait_model(args):
+    model = Sequential()
+    model.add(Conv1D(input_dim=len(args.inputs),
+                     input_length=args.window_size,
+                     filters=40,
+                     filter_length=16,
+                     border_mode='valid',
+                     activation="relu",
+                     init='normal'))
+
+    model.add(Dropout(0.3))
+    model.add(Conv1D(filters=64, kernel_size=16, activation="relu", init='normal', padding='valid'))
+    model.add(Dropout(0.3))
+    model.add(MaxPooling1D(pool_size=3, strides=3))
+    model.add(Flatten())
+
+    model.add(Dense(units=32, kernel_initializer='normal'))
+    model.add(Activation('relu'))
+
+    model.add(Dense(units=1, kernel_initializer=RandomNormal(mean=1.0, stddev=0.5, seed=None)))
+    model.add(Activation("relu"))
+
+    adamo = Adam(lr=0.001, beta_1=0.9, beta_2=0.999, epsilon=1e-08, clipnorm=1.)
+    my_metrics = [metrics.mean_squared_error, rmse_log]
+
+    model.compile(loss='mean_squared_error', optimizer=adamo, metrics=my_metrics)
+    model.summary()
+
+    return model
+
+
+def build_small_functional_bait_model(args):
+    bait_shape = (args.window_size, len(args.inputs),)
+    annotation_shape = (len(args.annotations),)
+
+    print("Bait shape: %s" % (bait_shape,))
+    input_baits = Input(name="bait", shape=bait_shape)
+
+    x = Conv1D(activation='relu',
+               padding='valid',
+               filters=100,
+               kernel_size=8,
+               kernel_initializer='normal', name='conv1')(input_baits)
+
+    x = Dropout(0.3)(x)
+
+    x = Conv1D(kernel_initializer="normal", activation="relu", padding="valid", filters=64, kernel_size=8)(x)
+    x = Dropout(0.3)(x)
+    x = MaxPooling1D(pool_size=3, strides=3)(x)
+
+    x = Conv1D(kernel_initializer="normal", activation="relu", padding="valid", filters=40, kernel_size=8)(x)
+    x = Dropout(0.3)(x)
+    x = MaxPooling1D(pool_size=3, strides=3)(x)
+
+    x = Flatten()(x)
+
+    x = Dense(units=32, kernel_initializer="normal", activation="relu")(x)
+
+    print("Annotation shape: %s" % (annotation_shape,))
+
+    input_annotations = Input(name="annotation", shape=annotation_shape)
+
+    xy = Concatenate(axis=-1)([x, input_annotations])
+
+    xy = Dense(units=32, kernel_initializer="normal", activation="relu")(xy)
+
+    predictions = Dense(units=1, kernel_initializer=RandomNormal(mean=1.0, stddev=0.5, seed=None), activation="relu")(
+        xy)
+    my_metrics = [metrics.mean_squared_error, rmse_log, gme]
+
+    # this creates a model that includes
+    # the Input layer and three Dense layers
+    model = Model(inputs=[input_baits, input_annotations], outputs=predictions)
+
+    # # add some TensorBoard annotations
+    # conv1d_1 = filter(lambda y: y.name == "conv1d_1", model.layers)[0]
```

Comment:
Why is this commented out?

Reply:
because I coudlnt' get it to work, but I wanted to show it to you to see if you could help!

Reply:
Ok, are you trying to visualize the weights in tensorboard or something more complicated?